### DISCUSSION TITLE
Differences of modes?

### DISCUSSION THREAD

**Discussion by maxchiron** [oobabooga/text-generation-webui#2586] [Q&A]
chat, chat-instruct, instruct
What's the difference???

**Comment by luthis1124** [comment]
I would love to get a clear answer on this one and how to use instruct.

**Comment by ossirytk** [comment]
There is a different prompt template used for chat and instruct. You can find the templates in char settings. The chat uses turn base template and the instruction has something like `Continue the chat dialogue below. Write a single reply for the character "<|character|>".`
You can edit these templates to get different kinds of output. I think Instruct is for when you want something like a GM in a tabletop roleplaying session. Where the template might be to give options on what will happen and then the adventurer chooses from those options. I have not really used the instruction modes too much so I might be completely off the base. I've used the chat-instruct type of prompts when running llama.cpp in terminal and they work pretty well with alpaca style roleplay models.
